Formula/p/pngquant.rb

```diff
@@ -1,14 +1,22 @@
 class Pngquant < Formula
   desc "PNG image optimizing utility"
   homepage "https://pngquant.org/"
-  url "https://pngquant.org/pngquant-2.18.0-src.tar.gz"
-  sha256 "e72194b52b36f040deaec49a1ddd5dcd8d4feecc3a5fe6c5e9589a9707b233d4"
+  url "https://static.crates.io/crates/pngquant/pngquant-3.0.2.crate"
+  sha256 "33f8501d8b81f34cb6f028a5d06772b9d7940e0bc2b15a5d0bce138cb74233cb"
   license :cannot_represent
-  head "https://github.com/kornelski/pngquant.git", branch: "master"
+  head "https://github.com/kornelski/pngquant.git", branch: "main"
 
   livecheck do
-    url "https://pngquant.org/releases.html"
-    regex(%r{href=.*?/pngquant[._-]v?(\d+(?:\.\d+)+)-src\.t}i)
+    url "https://crates.io/api/v1/crates/pngquant/versions"
```

Comment:
Thanks for bringing it up.

It would be straightforward to create a strategy for this. I discussed it with a contributor ~3 years ago (when they created a brew PR) but it was premature and we agreed to put it off until there was a clear benefit, enough usage, etc. [At the time, it would have only been used for `oakc` (which hadn't been merged yet) and it wasn't clear whether that formula would continue using a crate.]

This updated `livecheck` block will be duplicated for any crate check, so I think it makes sense to create a strategy now. The default behavior would work for the vast majority of crates and any outliers could be handled using a regex and/or `strategy` block (passing in the JSON and regex).

I'll put it together and create a brew PR but we can move forward with this `livecheck` block in the interim time.